docs/primehub-end-to-end-tutorial-4.md

```diff
@@ -10,45 +10,83 @@ description: Using PrimeHub from Training to Serving the Model
 </div>
 <br>
 
-In this tutorial, we will build a screw defect detection web app with the deployed model and [Streamlit](primehub-app-builtin-streamlit).
+In this tutorial, we will add a web interface to our deployed model using [Streamlit](primehub-app-builtin-streamlit). This will allow us to upload images, view results, and adjust the threshold for detecting a good or bad screw in a web browser.
 
-## Streamlit App
+## Before We Start
 
-Download [tutorial_screw_app.py](assets/tutorial_screw_app.py) and upload it to `~/<group_name>/screw`.
+Download [tutorial_screw_app.py](assets/tutorial_screw_app.py) to your local computer and then, in your Notebook Server, upload it to `~/<group_name>/screw`.
+
+
+## Install Streamlit
+
+
+Install Streamlit in PrimeHub Apps
 
-Install Streamlit to PrimeHub
 ![](assets/primehub-end-to-end-tutorial-web-app-1.png)
 
-Fill in `FILE_PATH` with `/project/phusers/screw/tutorial_screw_app.py`.
+On the Streamlit configuration page, enter **Screw Defect Detection** in the **Name** field.
+
+In the **Environment Variables** section, enter the following:
+
+For **FILE_PATH**, delete the default value, and then enter the path to the **tutorial_screw_app.py** we just uploaded, e.g.:
+
+ `/project/<group_name>/screw/tutorial_screw_app.py`
+
+Click **+ Add Field** to add a new **Environment Variable**.
+
+For the variable **name** enter **ENDPOINT**. For the variable **value** enter the Endpoint value we copied from our **Model Deployment** in the [previous section](primehub-end-to-end-tutorial-3#test-deployed-model).
 
-Add a new field `ENDPOINT` and fill in it with the deployed model endpoint in [3 - Manage, Compare, and Deploy the Model](primehub-end-to-end-tutorial-3).
 ![](assets/primehub-end-to-end-tutorial-web-app-2.png)
-![](assets/primehub-end-to-end-tutorial-web-app-3.png)
 
-Open your web app when it is ready.
+Click the Create button to finish installing Streamlit.
+
+## Using the Web Interface
+
+Once Streamlit is ready, click **Open** to launch the web interface.
+
 ![](assets/primehub-end-to-end-tutorial-web-app-4.png)
 
-Upload screw images from the upper-left corner. Then, the inference results are shown in the right side. You can adjust the threshold of good or bad screw. Also, you can select the column size for the result.
+Click the **Browse Files** button at the top left to upload images of screws. The uploaded images, along with their inference results, will be shown on the right.
+
 ![](assets/primehub-end-to-end-tutorial-web-app-5.png)
 
+Adjust the layout of the screw images by selecting a **Column Size** in the left sidebar. The threshold to determine good and bad screws can also be adjusted with the **Threshold** slider.
+
+Congratulations! We've finished the tutorial and completed our model deployment.
 
 ## Summary
```

Comment:
I love this part, Summary. 
My guess, we might have part 5, part 6 and so on in the future. I'd like to suggest to split "Summary" section into another page beneath the part4 so that we only need to maintain the Summary page.

Reply:
Good idea, this section is quite big. Bigger that the actual tutorial section for this part

Reply:
done